### ISSUE TITLE
Create script to allow conversion from HF ckpt to neox

### ISSUE THREAD

**Issue by Quentin-Anthony** [EleutherAI/gpt-neox#846]
Currently, we provide a script for converting a model checkpoint to HF (https://github.com/EleutherAI/gpt-neox/blob/main/tools/convert_to_hf.py), However, there is increasing need to convert a HF checkpoint back to gpt-neox (e.g. inference, building atop neox-based models on HF). 

We should create a reverse script to convert from HF back to the neox format. Whoever picks this up should do the following:
1. Read and understand our existing conversion script (https://github.com/EleutherAI/gpt-neox/blob/main/tools/convert_to_hf.py)
2. Choose a model with both neox and HF ckpt versions (Pythia would be good for this. Reach out to me or @haileyschoelkopf if you need access to the neox version of these ckpts)
3. Create a prototype HF-->neox script
4. Test the script under the following criteria:
    - [ ] Do downstream tasks perform the same among the baseline neox ckpt, the HF ckpt, and the converted HF-->neox ckpt?
    - [ ] Are logits the same among the baseline neox ckpt, the HF ckpt, and the converted HF-->neox ckpt?
    - [ ] Now that we can convert neox-->HF and HF-->neox, convert the same model back and forth several times. Do losses remain comparable?

**Comment by bentherien** [comment]
Happy to take this on. 

Thus far, I have completed steps 1, 2, and 3 resulting in a script that converts HF Pythia models to NeoX format and tests the logit difference between the HF and converted NeoX models on random inputs.   

While my conversion script works (i.e., the converted NeoX weights run), the logit testing reveals a small difference between the output by the HF and NeoX models. My hypothesis is that these differences occur due to stochasticity within NeoX or HF models that I'm unaware of.

Happy to share my script with someone that could shed light on this.

**Comment by StellaAthena** [comment]
@bentherien It sounds like you've made excellent progress. Why don't you open a PR with the data from your tests and we can take a look at it.

**Comment by bentherien** [comment]
@StellaAthena Sounds good, will make the PR later tonight or sometime tomorrow.

**Comment by bentherien** [comment]
I have opened #907

**Comment by StellaAthena** [comment]
The PR has been merged, so I’m closing this as complete.
